### PR TITLE
Improve CI Pipeline on PRs

### PR DESCRIPTION
**Pull request by Kidswiss**
## Summary

* Checks for label on PRs
* Builds images from the branch
* Open new PR on component-appcat with new version added
* Will auto tag the release on merge

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Update tests.
- [ ] Link this PR to related issues.

### CODE REVIEW

File: event.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  "action": "opened",
+  "pull_request": {
+    "merged": true,
+    "labels": [
+      {
+        "name": "bug"
+      }
+    ]
+  }
+}
```
Comment (wejdross) [inline event.json:11]: is this file necessary?

**Review by wejdross — COMMENTED** [review]
this branch needs rebasing :P 
do we have example pipeline somewhere where I can check how it behaves in real life?

**Comment by Kidswiss** [comment]
> do we have example pipeline somewhere where I can check how it behaves in real life?

This PR is already using it, you can check all the actions.

**Review by zugao — APPROVED** [review]
LGTM

**Comment by wejdross** [comment]
lgtm

**Comment by Kidswiss** [comment]
Can you have another look? I've added the cruft/cookiecutter config as well.

**Review by zugao — APPROVED** [review]
seems good
